pipelines/batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -157,13 +158,10 @@ static void runFhirFetch(FhirEtlOptions options, FhirContext fhirContext) throws
     EtlUtils.runPipelineWithTimestamp(pipeline, options);
   }
 
-  private static JdbcConnectionUtil createJdbcConnection(
+  private static DataSource createJdbcPooledDataSource(
       FhirEtlOptions options, DatabaseConfiguration dbConfig) throws PropertyVetoException {
-    return new JdbcConnectionUtil(
-        options.getJdbcDriverClass(),
-        dbConfig.makeJdbsUrlFromConfig(),
-        dbConfig.getDatabaseUser(),
-        dbConfig.getDatabasePassword(),
+    return JdbcConnectionPools.getPooledDataSource(
+        JdbcConnectionPools.dbConfigToDataSourceConfig(dbConfig),
         options.getJdbcInitialPoolSize(),
         options.getJdbcMaxPoolSize());
```

Comment:
Can these two methods `JdbcConnectionPools.getPooledDataSource` and `JdbcConnectionPools.dbConfigToDataSourceConfig` be merged into a single method in `JdbcConnectionPools` class?

Reply:
I am not sure I got this. Even in `FetchSearchPageFn`, I can still have `DataSource` returned once when required and create `jdbcWriter` out of it.

Reply:
Thanks for the clarification!